crypto/evp/evp_enc.c

```diff
@@ -365,8 +367,11 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
             ctx->num = 0;
             /* Don't reuse IV for CTR mode */
             if (iv != NULL) {
-                if ((n = EVP_CIPHER_CTX_get_iv_length(ctx)) <= 0)
+                n = EVP_CIPHER_CTX_get_iv_length(ctx);
+                if (n <= 0 || n > (int)sizeof(ctx->iv)) {
+                    ERR_raise(ERR_LIB_EVP, EVP_R_INVALID_IV_LENGTH);
                     return 0;
+                }
```

Comment:
Strange asymmetry: here a zero IV length is not allowed (and wasn't previously)?

Reply:
I believe this got mentioned before which is why I changed the previous check.

Reply:
This code is, however, inside an `iv != NULL` clause so a zero length would be really strange.

Reply:
Ah, I see!